pkg/socket/client.go

```diff
@@ -0,0 +1,62 @@
+package socket
+
+import (
+	"bufio"
+	"crypto/tls"
+	"net"
+	"time"
+)
+
+func NewSocket(config Config) *socket {
+	return &socket{
+		Config: config,
+		conn:   nil,
+	}
+}
+
+type socket struct {
+	Config
+	conn net.Conn
+}
+
+func (s *socket) Connect() (err error) {
+	if s.TlsConf == nil {
+		s.conn, err = net.DialTimeout(string(s.Network), s.Address, s.Timeout)
+	} else {
+		var d net.Dialer
+		d.Timeout = s.Timeout
+		s.conn, err = tls.DialWithDialer(&d, string(s.Network), s.Address, s.TlsConf)
+	}
+	return err
+}
+
+func (s *socket) Disconnect() error {
+	err := s.conn.Close()
+	s.conn = nil
+	return err
+}
+
+func (s *socket) Command(command string, process Processor) error {
+	if err := s.send(command, s.conn, s.Timeout); err != nil {
+		return err
+	}
+	return read(s.conn, process, s.Timeout)
+}
+
+func (s *socket) send(command string, writer net.Conn, timeout time.Duration) error {
+	if err := writer.SetWriteDeadline(time.Now().Add(timeout)); err != nil {
```

Comment:
Let's rename this method to "write". Also, it can be a function (I see we don't really use "*socket" there).

Reply:
done